docs/DeveloperGuide.md

```diff
@@ -4,35 +4,261 @@
+- `Command` : Executes the comman requested by the user.
+
+How the architecture components interact with each other
+
+The Sequence Diagram below shows how the components interact with each other for the scenario where the user issues the command `delete 1`.
+
+![Architecture Encode Data](images/sequence/Architecture_Encode_Data.png)
+*Figure 2: Architecture Encode Sequence Diagram*
+
+### Ui Component
+API: [Ui.java]({repoURL}src/main/java/seedu/cafectrl/ui/Ui.java)
+
+![Ui Class Diagram](images/class/Ui.png)
+<br>*Figure 3: Ui Class Diagram*
+
+The `Ui` component is responsible for interacting with the user. Within CafeCtrl, `Ui` is instantiated by `Parser`, `Command`, `Main`, `Data`, and `Storage` components to access the print methods in `Ui.java`.
+
+In the Ui component,
+- `Ui.java` consists of multiple methods that received the user input and prints messages to the system console for users to see
+- `Messages.java` consists of multiple strings that contains greeting, command, and goodbye messages to be shown to user
+- `ErrorMessages.java` consists of multiple strings that contain error messages to be shown to user when an incorrect command or exception has been returned
+
+### Parser Component
+API: [Parser.java]({repoURL}src/main/java/seedu/cafectrl/parser/Parser.java)
+
+![Parser Class Diagram](images/class/Parser.png)
+
+*Figure 4: Parser Class Diagram*
+
+Note that `CafeCtrl` only have access to the interface `ParserUtil` although the run-time type object is `Parser`. With this, we are able to decrease coupling between `CafeCtrl` and `Parser`, allowing for easier maintenance. This also ensures the testability as we could provide mock or stub dependencies during testing, we could isolate the behavior of the class and focus on unit testing without external dependencies. 
+
+![Parser Parsing User Input Sequence Diagram](images/sequence/Parser.png)
+
+*Figure 5: Parser Parsing User Input Sequence Diagram*
+
+When user input a string to `Main`,  it passes the full user input to `Parser` via `parseCommand`. In `parseCommand`,  it finds the matching keyword for different command from the user input, then it calls the respective `prepareCommand` method within `Parser`. `prepareCommand` then generates the corresponding command class and return it to `parseCommand`, which returns the `Command` back to `Main` for execution.
+
+### Storage Component
+API: [Storage.java]({repoURL}src/main/java/seedu/cafectrl/storage/Storage.java)
+
+![Storage Class Diagram](images/class/Storage.png)
+
+*Figure 6: Storage Class Diagram*
+
+The `Storage` class,
+- loads and saves the list of dishes on the `Menu`, available ingredient stock in `Pantry` and orders for the day in `OrderList` in a text file.
+- depends on `Menu`, `Pantry` and `Sales` objects (which are found in the data package).
+- is composed of `FileManager` object as the text file needs to be located first before reading or writing.
+
+### Data Component
+API: []
```

Comment:
What is this supposed to mean?

It confuses me and I think you should clarify this 
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/49584216/821da2df-07aa-4445-887a-71f615f16b2f)

Reply:
We forgot to add hyperlink here! It is updated in the latest version